common/persistence/execution_manager_test.go

```diff
@@ -480,6 +480,159 @@ func TestExecutionManager_UpdateWorkflowExecution(t *testing.T) {
 	assert.Equal(t, stats, res.MutableStateUpdateSessionStats)
 }
 
+func TestSerializeWorkflowSnapshot(t *testing.T) {
+	for _, tc := range []struct {
+		name         string
+		prepareMocks func(*MockPayloadSerializer)
+		input        *WorkflowSnapshot
+		checkRes     func(*testing.T, *InternalWorkflowSnapshot, error)
+	}{
+		{
+			name: "success",
+			prepareMocks: func(mockedSerializer *MockPayloadSerializer) {
+				mockedSerializer.EXPECT().SerializeEvent(gomock.Any(), gomock.Any()).Return(sampleEventData(), nil).Times(5)
+				mockedSerializer.EXPECT().SerializeResetPoints(gomock.Any(), gomock.Any()).Return(sampleResetPointsData(), nil).Times(1)
+				mockedSerializer.EXPECT().SerializeVersionHistories(gomock.Any(), gomock.Any()).Return(sampleTestCheckSumData(), nil).Times(1)
+				mockedSerializer.EXPECT().SerializeChecksum(gomock.Any(), gomock.Any()).Return(NewDataBlob([]byte("test-checksum"), common.EncodingTypeThriftRW), nil).Times(1)
+			},
+			input: sampleWorkflowSnapshot(),
+			checkRes: func(t *testing.T, res *InternalWorkflowSnapshot, err error) {
+				assert.NoError(t, err)
+				assert.NotNil(t, res)
+			},
+		},
+		{
+			name: "nil info",
+			prepareMocks: func(mockedSerializer *MockPayloadSerializer) {
+				mockedSerializer.EXPECT().SerializeChecksum(gomock.Any(), gomock.Any()).Return(sampleTestCheckSumData(), nil).Times(1)
+			},
+			input: &WorkflowSnapshot{},
+			checkRes: func(t *testing.T, res *InternalWorkflowSnapshot, err error) {
+				assert.NoError(t, err)
+				assert.Equal(t, &InternalWorkflowSnapshot{
+					ExecutionInfo:       &InternalWorkflowExecutionInfo{},
+					ChecksumData:        sampleTestCheckSumData(),
+					StartVersion:        common.EmptyVersion,
+					LastWriteVersion:    common.EmptyVersion,
+					ActivityInfos:       make([]*InternalActivityInfo, 0),
+					ChildExecutionInfos: make([]*InternalChildExecutionInfo, 0),
+				}, res)
+			},
+		},
+		{
+			name: "serialize event error",
+			prepareMocks: func(mockedSerializer *MockPayloadSerializer) {
+				mockedSerializer.EXPECT().SerializeEvent(gomock.Any(), gomock.Any()).Return(nil, assert.AnError).Times(1)
+			},
+			input: sampleWorkflowSnapshot(),
+			checkRes: func(t *testing.T, res *InternalWorkflowSnapshot, err error) {
+				assert.Error(t, err)
+				assert.Nil(t, res)
+			},
+		},
+		{
+			name: "serialize points error",
+			prepareMocks: func(mockedSerializer *MockPayloadSerializer) {
+				mockedSerializer.EXPECT().SerializeEvent(gomock.Any(), gomock.Any()).Return(sampleEventData(), nil).Times(1)
+				mockedSerializer.EXPECT().SerializeResetPoints(gomock.Any(), gomock.Any()).Return(nil, assert.AnError).Times(1)
+			},
+			input: sampleWorkflowSnapshot(),
+			checkRes: func(t *testing.T, res *InternalWorkflowSnapshot, err error) {
+				assert.Error(t, err)
+				assert.Nil(t, res)
```

Comment:
We should check that this is the error we expect
We should not assert nil if we expect an error the return value does not matter

Same comment for all the cases below